libs/vkd3d/state.c

```diff
@@ -3353,44 +3353,59 @@ static HRESULT vkd3d_bindless_state_add_binding(struct vkd3d_bindless_state *bin
     VkDescriptorSetLayoutBindingFlagsCreateInfoEXT vk_binding_flags_info;
     const struct vkd3d_vk_device_procs *vk_procs = &device->vk_procs;
     VkDescriptorSetLayoutCreateInfo vk_set_layout_info;
-    VkDescriptorSetLayoutBinding vk_binding_info;
-    VkDescriptorBindingFlagsEXT vk_binding_flags;
+    VkDescriptorSetLayoutBinding vk_binding_info[9];
+    VkDescriptorBindingFlagsEXT vk_binding_flags[9];
+    VkDescriptorSetLayoutBinding *vk_binding;
+    unsigned int i;
     VkResult vr;
 
     set_info->vk_descriptor_type = vk_descriptor_type;
     set_info->heap_type = flags & VKD3D_BINDLESS_SET_SAMPLER
             ? D3D12_DESCRIPTOR_HEAP_TYPE_SAMPLER
             : D3D12_DESCRIPTOR_HEAP_TYPE_CBV_SRV_UAV;
     set_info->flags = flags;
-    set_info->binding_index = 0;
+    set_info->binding_index = vkd3d_popcount(flags & VKD3D_BINDLESS_SET_EXTRA_MASK);
```

Comment:
Will flags be 0 for non-SSBO descriptor types?

Reply:
What do you mean exactly?

`flags & VKD3D_BINDLESS_SET_EXTRA_MASK` will be 0 whenever there are no extra bindings set. `flags` is never going to be 0 since that'll contain `VKD3D_BINDLESS_SET_UAV` and friends.

Reply:
I mean if `flags & VKD3D_BINDLESS_SET_EXTRA_MASK` is only non-null for one descriptor type?

Reply:
Ok, I think I see now.